### DISCUSSION TITLE
Not seeing Rosetta Model Explorer

### DISCUSSION THREAD

**Discussion by stevevestal** [opencaesar/platform#29] [Q&A]
I followed the instructions in https://www.opencaesar.io/oml-tutorials/#getting-ready.  The instructions say I should see a Model Explorer in the upper-left, but I see the usual Project Explorer. Is this the Sirius Model Explorer (only Model Explorer in the list)?  Or does the most recent version use the default Project Explorer?

**Comment by stevevestal** [comment]
The tutorial is a tiny bit outdated.  In Project Explorer, Create a project.., in folder OML select OML Project.

**Reply by stevevestal** [reply]
for
```
vocabulary <http://example.com/project/tutorial1/vocabulary/pizza#> as pizza {
}
```
I get the error
```
Vocabulary iri 'http://example.com/project/tutorial1/vocabulary/pizza' does not resolve to 'platform:/resource/tutorial1/src/oml/example.com/project/pizza.oml' using the catalog	pizza.oml	/tutorial1/src/oml/example.com/project
```
I tried several variations.  The OML Specification says a NAMESPACE is an IRI.  Are there restrictions on IRIs?

**Comment by melaasar** [comment]
Did you switch to the Modeling Perspective as instructed?
```
Run OML Rosetta
1. Once Rosetta opens with the workspace, switch to the Modeling Perspective.
```

**Reply by stevevestal** [reply]
Thanks, that did it.

When I got to 1.3 Create OML Vocabulary, step 2, and pushed finish, it created the pizza.oml file but not a vocabulary subfolder to contain it.  When I manually created a vocabulary subfolder and moved the pizza.oml file there, the "Vocabulary iri ... does not resolve ..." error disappeared.   Is the NAMESPACE IRI required to match the subfolder nesting of the OML model?  I did a hasty search of the OML Specification and didn't find a constraint like that on NAMESPACE IRI.

**Comment by melaasar** [comment]
You are right. The fact that the wizard failed to create the subfolder `vocabulary` is a bug (a recent regression). I will fix this and release a patch soon. Meanwhile, what you did is the correct thing. The catalog.xml is what is used to resolve IRIs into Paths. The default rule that got generated by the wizard, i.e., 
```
	<rewriteURI uriStartString="http://example.com/tutorial1/" rewritePrefix="src/oml/example.com/tutorial1/"/>
```

Is what requires the folder paths to match the IRI path. This is chosen by default for simplicity. However, one could define a different mapping rule in the catalog.
